DokuwikiXMLExport.php

```diff
@@ -50,71 +49,117 @@ class DokuwikiXMLExport
+        $this->conf = $conf;
+        $this->pages = $this->getPageIds();
+    }
+
+    /**
+     * Returns all pageIds, excluding those who were set in the configuration.
      *
-     * @param $start integer Determines the first item (offset) to be exported.
-     * @param $count integer Determines the interval size / number of items to be exported.
-     * @return string Returns the XML.
+     * @return array pageIds.
      */
-    public function generateXMLExport($start, $count)
+    private function getPageIds()
     {
         $indexer = new Doku_Indexer();
         $pagesAndDeletedPages = $indexer->getPages();
 
         $excludedPages = $this->splitConfigToArray($this->conf['plugin']['findologicxmlexport']['excludePages']);
-        $pages = null;
+
         foreach ($pagesAndDeletedPages as $page) {
             if (p_get_metadata($page)['description']) { // Only get pages with content
                 if (!in_array($page, $excludedPages)) { // Exclude pages from config
-                    $pages[] = $page;
+                    $ids[] = $page;
                 }
             }
         }
-        $exporter = Exporter::create(Exporter::TYPE_XML, $count);
 
-        $total = count($pages);
+        return $ids;
+    }
 
-        // The count can't be higher then the total number of pages.
-        $count = min($total, $count);
+    /**
+     * Formats Config string to an array.
+     *
+     * @param string $config Excluded pages in a string.
+     * @return array Returns the pages that should be excluded as array.
+     */
+    private function splitConfigToArray($config)
+    {
+        $array = explode(',', $config);
+        $trimmedArray = array_map('trim', $array);
+        return $trimmedArray;
+    }
 
+    /**
+     * Generate the entire XML Export based on the Dokuwiki metadata.
+     *
+     * @param $start integer Determines the first item (offset) to be exported.
+     * @param $count integer Determines the interval size / number of items to be exported.
+     * @return string Returns the XML.
+     */
+    public function generateXMLExport($start, $count)
+    {
+        $exporter = Exporter::create(Exporter::TYPE_XML, $count);
+
+        $total = count($this->pages);
+        $count = min($total, $count); // The count can't be higher then the total number of pages.
 
         // If count + start is higher then total, then something went totally wrong.
         if (($count + $start) > $total) {
             throw new \InvalidArgumentException("Error: Failed while trying to validate start and count values");
         }
 
         $items = array();
-        for ($i = $start; $i < $start + $count; $i++) {
-            $item = $exporter->createItem($i);
+        for ($this->key = $start; $this->key < $start + $count; $this->key++) {
+            $item = $exporter->createItem($this->key);
 
             $name = new Name();
-            $name->setValue($this->getName($pages, $i));
+            $name->setValue($this->getName());
```

Comment:
I'd suggest something like `$this->getName($this->pages[$key]`). The `getName()` method should not have to worry about where to find the page, but just about extracting the name from a page. This applies to all similar methods as well.

Reply:
This is what you've meant. I was a bit confused about how the behavior should be. But I am totally on your side this makes sense.